### PR TITLE
feat(BigNumbers): new component BigNumbers

### PR DESCRIPTION
**Pull request by jlongshore**
Contributes to [issue 4060](https://github.com/carbon-design-system/ibm-products/issues/4060)

Migrate **ICA** from `Products v1 / Security / ICA` to **Products v2** as `BigNumber`. 

`v1` References: [GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/ICA/ICA.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-ica--default).

#### What did you change?

- Name change (ICA -> BigNumbers)
- Removing Numerical package - switched to using [JS' NumberFormat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat)
- [Previously documented](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/ICA/README.md) locale ‘chs‘ (Simplified Chinese) is not supported
- Removed the ICASkeleton component in favor of a `loading` prop (see below in prop updates)

**Prop updates:**
| Old prop | New prop | Change |
|-|-|-|
| _className_ | _className_ | _Unchanged_|
|_forceShowTotal_|_forceShowTotal_|_Unchanged_|
| — | fractionDigits | **Number** : Optional value to control the maximum fraction digits used when truncating the value and total. |
|_iconButton_|_iconButton_|_Unchanged_|
|_information_| tooltipDescription | **String**: Label applied to a Tooltip - marked with the Information icon. |
|_label_|_label_|_Unchanged_|
| — | loading | **Boolean** :  Specify if the BigNumbers is in a loading state - shows component in a skeleton fashion |
|_locale_|_locale_|_Unchanged_|
|_percentage_|_percentage_|_Unchanged_|
|_size_|_size_|_Unchanged_|
|_total_|_total_|_Unchanged_|
|_trending_|_trending_|_Unchanged_|
|_truncate_|_truncate_|_Unchanged_|
|_value_|_value_|_Unchanged_|

#### How did you test and verify your work?
- [x] Storybook review.
- [x] Tests migrated and added to.
- [x] Design review by Sarah Pedregon.

### CODE REVIEW

File: packages/core/story-structure.js
Comment (elycheea) [inline packages/core/story-structure.js:37]: Sentence case!

```suggestion
          { n: 'Big numbers', s: ['c/BigNumbers'] },
```

File: packages/ibm-products/src/components/BigNumbers/utils/enums.js
Comment (davidmenendez) [inline packages/ibm-products/src/components/BigNumbers/utils/enums.js:1]: i like the idea here, but i think you can just name this file `constants`. typically for a `utils` folder i'd expect to see utility functions here. if you wanted to move some of those functions here like `truncateValue` and `formatValue` that would be a great way to utilize a utils file and clean up the jsx

**Review by davidmenendez — CHANGES_REQUESTED** [review]
looks great! have a few comments and suggestions. also i think for the sake of keeping things clean you can remove a lot of the explanatory comments from the main `BigNumbers.js` file. anything that's particularly important should probably just be noted in the readme

File: packages/core/story-structure.js
Comment (jlongshore) [reply packages/core/story-structure.js:37]: old habits die hard, good catch

File: packages/ibm-products/src/components/BigNumbers/BigNumbers.js
Comment (jlongshore) [reply packages/ibm-products/src/components/BigNumbers/BigNumbers.js:237]: Good catch!

File: packages/ibm-products/src/components/BigNumbers/BigNumbers.js
Comment (jlongshore) [reply packages/ibm-products/src/components/BigNumbers/BigNumbers.js:71]: Or maybe we use the description (something like `tooltipDescription`) seems to communicate that **to me** better - although I see wherever we use `<Tooltip...`  in Products, the prop we use is `label`. [Tooltip storybook link](https://react.carbondesignsystem.com/?path=/story/components-tooltip--default) for reference.

File: packages/ibm-products/src/components/BigNumbers/BigNumbers.js
Comment (jlongshore) [reply packages/ibm-products/src/components/BigNumbers/BigNumbers.js:126]: Totally agree!

File: packages/ibm-products/src/components/BigNumbers/utils/enums.js
Comment (jlongshore) [reply packages/ibm-products/src/components/BigNumbers/utils/enums.js]: Yep, I follow. Advise if this needs to be in a directory or if alongside the component in a `utils.js` file will work as I have it now. Formatted that file with constants at the top and utility functions below... seemed to me to clean up the component file some.

File: packages/ibm-products/src/components/BigNumbers/BigNumbers.js
Comment (davidmenendez) [reply packages/ibm-products/src/components/BigNumbers/BigNumbers.js]: `tooltipDescription` is good!

File: packages/ibm-products/src/components/BigNumbers/utils/enums.js
Comment (davidmenendez) [reply packages/ibm-products/src/components/BigNumbers/utils/enums.js]: if you wanted to just create a `constants.js` and import it into the file i think that would be sufficient

**Review by davidmenendez — DISMISSED** [review]

File: packages/core/story-structure.js
Comment (elycheea) [inline packages/core/story-structure.js:66]: I wonder if this little space is why you got that merge conflict. 😅

Comment (elycheea) [reply packages/core/story-structure.js:66]: Let’s fix string formatter to sentence case and alphabetical since I think `String formatter` should come before `Tag set` anyway.

**Review by elycheea — APPROVED** [review]
Re-approving after merge conflict. 💁‍♀️

**Pull request merged**
